### PR TITLE
Adds fastboot support

### PR DESCRIPTION
**Pull request by Duder-onomy**
Avoid registering events when in fastboot.

### CODE REVIEW

File: addon/services/unified-event-handler.js [hunk 1]
```diff
@@ -56,8 +61,10 @@ export default Ember.Service.extend(Ember.Evented, {
    * @return {Void}
    */
   register(target, eventName, callback) {
-    let handlerInfo = this._registerDOMHandler(target, eventName);
-    this._registerEmberHandler(handlerInfo, callback);
+    if (!this.get('isFastBoot')) {
```
Comment (trentmwillis) [inline addon/services/unified-event-handler.js]: Both here and below, can we use a return-early pattern instead?

```js
if (this.get('isFastBoot')) {
  return;
}
```

It's easier to parse the flow of the logic IMO.

File: addon/services/unified-event-handler.js [hunk 2]
```diff
@@ -47,6 +47,11 @@ export default Ember.Service.extend(Ember.Evented, {
     this._throttledEventTimers = [];
   },
 
+  isFastBoot: Ember.computed(function() {
+    const fastbootService = Ember.getOwner(this).lookup('service:fastboot');
```
Comment (trentmwillis) [inline addon/services/unified-event-handler.js]: Pardon my ignorance, but is this the recommended way to check if you're in FastBoot? Seems kind of roundabout, but I haven't done much with FastBoot.

**Comment by trentmwillis** [comment]
Also, looks like CI is failing for some reason: https://github.com/trentmwillis/ember-singularity/issues/22. Will try to investigate this weekend so that we can merge this when ready.

File: addon/services/unified-event-handler.js (hunk 1 above)
Comment (Duder-onomy) [reply addon/services/unified-event-handler.js]: No problem, will update.

File: addon/services/unified-event-handler.js (hunk 2 above)
Comment (Duder-onomy) [reply addon/services/unified-event-handler.js]: Yes, I believe this is the recommended way to check for fastboot:
https://www.ember-fastboot.com/docs/addon-author-guide#accessing-the-fastboot-service

I am willing to change it if you feel diff.

**Comment by Duder-onomy** [comment]
That Travis failure is related to: #9024
I will PR a fix for that separately.

File: addon/services/unified-event-handler.js (hunk 2 above)
Comment (trentmwillis) [reply addon/services/unified-event-handler.js]: Seems good to me then. Thanks for the reference.

**Comment by trentmwillis** [comment]
@Duder-onomy I've merged the other PR, if you can rebase this, the tests should pass and then I'll merge

**Comment by Duder-onomy** [comment]
Boom!. @trentmwillis Rebased.

**Comment by trentmwillis** [comment]
Thanks! I'll publish another release tonight 🙂

**Comment by trentmwillis** [comment]
@Duder-onomy I've published version 1.1.0 with all your recent contributions. Thanks a bunch!

**Pull request merged**